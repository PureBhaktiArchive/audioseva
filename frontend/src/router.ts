/*
 * sri sri guru gauranga jayatah
 */

import firebase from "firebase/app";

import Vue from "vue";
import Router from "vue-router";

Vue.use(Router);

export const router = new Router({
  mode: "history",
  routes: [
    { path: "*", redirect: "/" },
    {
      path: "/login",
      meta: { guestOnly: true },
      component: () => import("@/views/Login.vue")
    },
    {
      path: "/",
      meta: { requireAuth: true },
      component: () => import("@/views/Dashboard.vue"),
      children: [
        {
          path: "",
          component: () => import("@/views/Home.vue")
        },
        {
          path: "cr/allot",
          component: () => import("@/views/CRAllotment.vue")
        },
        {
          path: "sqr/",
          component: () => import("@/views/SQR.vue"),
          meta: { activator: true, activatorName: "Sound Quality Reporting", menuIcon: "fas fa-headphones" },
          children: [
            { path: "",
              component: () => import("@/views/SQRFiles.vue"),
              meta: { menuItem: true}
            },
            {
              path: "allot",
              component: () => import("@/views/SQRAllotment.vue"),
              meta: { menuItem: true }
            },
            { path: "statistics",
              component: () => import("@/views/SQRFileStatistics.vue"),
              meta: { menuItem: true }
            }
          ]
        },
        {
          path: "se/",
          component: () => import("@/views/SE/SE.vue"),
          meta: { activator: true, activatorName: "Sound Engineering", menuIcon: "fas fa-music" },
          children: [
            { path: "", component: () => import("@/views/SE/Tasks.vue"), meta: { menuItem: true } },
            { path: "allot", component: () => import("@/views/SE/Allotment.vue"), meta: { menuItem: true } }
          ]
        },
        {
          path: "te/allot",
          component: () => import("@/views/TEAllotment.vue")
        },
        {
          path: "te/fc/allot",
          component: () => import("@/views/TFCAllotment.vue")
        },
        {
          path: "/qc/allot",
          component: () => import("@/views/QCAllotment.vue")
        }
      ]
    }
  ]
});

router.beforeEach((to, from, next) => {
  let currentUser = firebase.auth().currentUser;
  let requireAuth = to.matched.some(record => record.meta.requireAuth);
  let guestOnly = to.matched.some(record => record.meta.guestOnly);

  if (requireAuth && !currentUser)
    next({
      path: "/login",
      query: { redirect: to.fullPath }
    });
  else if (guestOnly && currentUser) next("/");
  else next();
});
