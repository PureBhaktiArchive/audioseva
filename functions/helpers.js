const functions = require('firebase-functions');
let SibApiV3Sdk = require('sib-api-v3-sdk');
const sendInBlueSecretKey = functions.config().audioseva.send_in_blue.key;
const defaultClient = SibApiV3Sdk.ApiClient.instance;
let apiKey = defaultClient.authentications['api-key'];
apiKey.apiKey = sendInBlueSecretKey;



exports.checkValidMP3 = filePath => (filePath.startsWith("mp3/") && filePath.endsWith(".mp3"))


exports.removeFromDB = (db, dbPath) => {
    let ref = db.ref(dbPath);
    ref.remove()
        .then(() => console.log("Deleted."))
        .catch(error => console.log(error));
}


exports.sendEmail = (to, bcc, templateId, params) => {
    let apiInstance = new SibApiV3Sdk.SMTPApi();
    sendEmail = new SibApiV3Sdk.SendSmtpEmail();

    sendEmail = {
        to: [{ email: to }], // add a name prop if it fails to send
        bcc,
        templateId,
        params
    };

    apiInstance.sendTransacEmail(sendEmail).then(function(data) {
        console.log('API called successfully. Returned data: ');
        console.log(data);
        return 1;
    }, err => console.log(err)).catch(err => console.log(err));
}
