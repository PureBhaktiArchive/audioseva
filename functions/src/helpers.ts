const functions = require('firebase-functions');
let SibApiV3Sdk = require('sib-api-v3-sdk');
const sendInBlueSecretKey = functions.config().send_in_blue.key;
const defaultClient = SibApiV3Sdk.ApiClient.instance;
let apiKey = defaultClient.authentications['api-key'];
apiKey.apiKey = sendInBlueSecretKey;



export const checkValidMP3 = filePath => (filePath.startsWith("mp3/") && filePath.endsWith(".mp3"))

export const createMP3DBRef = (filePath, _module) => {
    const parts = filePath.split('/');
        
    const list = parts[1];
    const mp3 = parts[2];
    let file_name = mp3.slice(0, -4);

    return `/${_module}/files/${list}/${file_name}`;
}

export const extractListFromFilename = (fileName) => {
    return fileName.match(/^[^-]*[^ -]/g)[0];
}


export const storeFileNameToDB = (filePath, db, _module) => {
    if(checkValidMP3(filePath)) {
        let ref = db.ref(createMP3DBRef(filePath, _module));

        //checking if the file already exists in the RT db
        ref.child("status").once('value')
        .then(snapshot => {
            if(!snapshot.exists())
                ref.set({status: "Spare"});
            return 1;
        }).catch(err => console.log(err));
    }
}

export const removeFromDB = (db, dbPath) => {
    let ref = db.ref(dbPath);
    ref.remove()
        .then(() => console.log("Deleted."))
        .catch(error => console.log(error));
}


export const sendEmail = (to, bcc, templateId, params) => {
    let apiInstance = new SibApiV3Sdk.SMTPApi();
    let sendEmail = new SibApiV3Sdk.SendSmtpEmail();

    sendEmail = {
        to: [{ email: to }], // add a name prop if it fails to send
        bcc,
        templateId,
        params
    };

    apiInstance.sendTransacEmail(sendEmail).then(function(data) {
        // Message sending result
        console.log(data);
        return 1;
    }, err => console.log(err)).catch(err => console.log(err));
}
