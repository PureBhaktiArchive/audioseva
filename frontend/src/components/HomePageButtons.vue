<template>
  <div>
    <v-btn class="mx-2" :key="`${button.text}-${index}`" v-for="(button, index) in routeButtons" v-bind="button.props">
      {{ button.text }}
    </v-btn>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import { RouteConfig } from "vue-router";
import { getHomePageRoutes } from "@/router";

@Component({
  name: "HomePageButtons"
})
export default class HomePageButtons extends Vue {
  userClaims: any = {};
  routeButtons: any[] = [];

  async mounted() {
    this.makeButtons(await getHomePageRoutes());
  }

  makeButtons(routes: RouteConfig[], parentPath: string = "/") {
    const defaultButtonProps = {
      color: "primary"
    };
    routes.forEach(route => {
      const fullPath = `${parentPath}${route.path}`;
      if (route.meta.homePageLink) {
        this.routeButtons.push({
          text: route.meta.homePageLink.text,
          props: {
            ...defaultButtonProps,
            ...route.meta.homePageLink.props,
            to: fullPath
          }
        });
      }
      if (route.children) {
        this.makeButtons(route.children, fullPath);
      }
    });
  }
}
</script>

<style scoped>
</style>
