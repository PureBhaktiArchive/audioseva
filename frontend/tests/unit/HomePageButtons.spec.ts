import Vue from "vue";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import VueRouter from "vue-router";
import flushPromises from "flush-promises";
import HomePageButtons from "@/components/HomePageButtons.vue";
import { router } from "@/router";
import { mockClaims } from "./helpers";

describe("HomePageButtons", () => {
  let localVue: typeof Vue;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(VueRouter);
  });

  test.each`
    claims
    ${{ TE: { editor: true } }}
    ${{ TE: { coordinator: true } }}
  `("should render buttons that match claims $claims", async ({ claims }) => {
    await mockClaims(claims);
    const wrapper = shallowMount(HomePageButtons, {
      localVue,
      router
    });
    await flushPromises();
    expect((wrapper.vm as any).routeButtons).toMatchSnapshot();
  });
});
