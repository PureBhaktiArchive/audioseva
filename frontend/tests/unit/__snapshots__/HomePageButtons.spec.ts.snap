// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageButtons should render buttons that match claims {"TE": [Object]} 1`] = `
Array [
  Object {
    "props": Object {
      "color": "primary",
      "to": "/te/upload",
    },
    "text": "Upload",
  },
]
`;

exports[`HomePageButtons should render buttons that match claims {"TE": [Object]} 2`] = `
Array [
  Object {
    "props": Object {
      "color": "primary",
      "to": "/te/tasks",
    },
    "text": "TE",
  },
  Object {
    "props": Object {
      "color": "primary",
      "to": "/te/upload",
    },
    "text": "Upload",
  },
]
`;
